src/sentry/rules/history/backends/postgres.py

```diff
@@ -42,15 +46,20 @@ def fetch_rule_groups_paginated(
         cursor: Cursor | None = None,
         per_page: int = 25,
     ) -> CursorResult[Group]:
+        filtered_history = RuleFireHistory.objects.filter(
+            rule=rule,
+            date_added__gte=start,
+            date_added__lt=end,
+        )
+
+        group_max_dates = filtered_history.filter(group=OuterRef("group")).order_by("-date_added")[
+            :1
+        ]
         qs = (
-            RuleFireHistory.objects.filter(
-                rule=rule,
-                date_added__gte=start,
-                date_added__lt=end,
-            )
-            .select_related("group")
-            .values("group")
-            .annotate(count=Count("id"), last_triggered=Max("date_added"))
+            filtered_history.values("group")
+            .annotate(count=Count("group"))
+            .annotate(event_id=Subquery(group_max_dates.values("event_id")))
+            .annotate(last_triggered=Subquery(group_max_dates.values("date_added")))
```

Comment:
Not sure how to do this without subqueries, let me know if there's a better way or schema